torchvision/prototype/datasets/decoder.py

```diff
@@ -13,3 +16,17 @@ def raw(buffer: io.IOBase) -> torch.Tensor:
 
 def pil(buffer: io.IOBase, mode: str = "RGB") -> torch.Tensor:
     return pil_to_tensor(PIL.Image.open(buffer).convert(mode.upper()))
+
+
+def av_kf(buffer: io.IOBase, **read_video_kwargs: Any) -> Dict[str, Any]:
```

Comment:
I feel like a decoder in this file should follow the "one file handle, one decoded output" paradigm. That means, I would prefer to have an `av` function here that decodes the whole video. Given that you also don't use this function in the video utilities that you wrote, I think that should not be a problem.

Reply:
actually, after thinking about it, all decoding is atm happening in the `video_utils` file so I'll revert this file since we don't use it

Reply:
Fine by me. We can always add it later if the need arises.